src/main/java/org/cryptomator/common/mountpoint/CustomMountPointChooser.java

```diff
@@ -60,21 +70,47 @@ public boolean prepare(Volume caller, Path mountPoint) throws InvalidMountPointE
 				throw new InvalidMountPointException(new IllegalStateException("Not implemented"));
 			}
 		}
-		LOG.debug("Successfully checked custom mount point: {}", mountPoint);
-		return false;
 	}
 
-	private void prepareParentNoMountPoint(Path mountPoint) throws InvalidMountPointException {
-		//This the case on Windows when using FUSE
-		//See https://github.com/billziss-gh/winfsp/issues/320
-		Path parent = mountPoint.getParent();
-		if (!Files.isDirectory(parent)) {
-			throw new InvalidMountPointException(new NotDirectoryException(parent.toString()));
-		}
-		//We must use #notExists() here because notExists =/= !exists (see docs)
-		if (!Files.notExists(mountPoint, LinkOption.NOFOLLOW_LINKS)) {
-			//File exists OR can't be determined
-			throw new InvalidMountPointException(new FileAlreadyExistsException(mountPoint.toString()));
+	//This the case on Windows when using FUSE
+	//See https://github.com/billziss-gh/winfsp/issues/320
+	void prepareParentNoMountPoint(Path mountPoint) throws InvalidMountPointException {
+		Path hideaway = getHideaway(mountPoint);
+		var mpExists = Files.exists(mountPoint);
+		var hideExists = Files.exists(hideaway);
+
+		//both resources exist (whatever type)
+		//TODO: possible improvement by just deleting an _empty_ hideaway
+		if (mpExists && hideExists) {
+			throw new InvalidMountPointException(new FileAlreadyExistsException(hideaway.toString()));
+		} else if (!mpExists && !hideExists) { //neither mountpoint nor hideaway exist
+			throw new InvalidMountPointException(new NoSuchFileException(mountPoint.toString()));
+		} else if (!mpExists) { //only hideaway exists
+
+			if (!Files.isDirectory(hideaway)) {
+				throw new InvalidMountPointException(new NotDirectoryException(hideaway.toString()));
+			}
+			LOG.info("Mountpoint {} for winfsp mount seems to be not properly cleaned up. Will be fixed on unmount.", mountPoint);
+			try {
+				Files.setAttribute(hideaway, "dos:hidden", true);
+			} catch (IOException e) {
+				throw new InvalidMountPointException(e);
+			}
+		} else {
+			if (!Files.isDirectory(mountPoint)) {
+				throw new InvalidMountPointException(new NotDirectoryException(mountPoint.toString()));
+			}
+			try {
+				if (Files.list(mountPoint).findFirst().isPresent()) {
+					throw new InvalidMountPointException(new DirectoryNotEmptyException(mountPoint.toString()));
+				}
```

Comment:
this leaks an unclosed stream returned by `Files.list(mountPoint)`

```suggestion
				try (var stream = Files.list(mountPoint)) {
					if (stream.findAny().isPresent()) {
						throw new InvalidMountPointException(new DirectoryNotEmptyException(mountPoint.toString()));
					}
				}
```

Reply:
Already fixed in [fba0df1](https://github.com/cryptomator/cryptomator/pull/2082/commits/fba0df10f9503f0460aa0311d09de6710f13b9f9)